src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.Dsa.cs

```diff
@@ -101,24 +101,25 @@ internal static DSAParameters ExportDsaParameters(SafeDsaHandle key, bool includ
                 throw new CryptographicException();
             }
 
-            IntPtr p_bn, q_bn, g_bn, y_bn, x_bn; // these are not owned
+            SafeBignumHandle p_bn, q_bn, g_bn, y_bn, x_bn;
             int    p_cb, q_cb, g_cb, y_cb, x_cb;
 
-            bool refAdded = false;
-            try
+            if (!GetDsaParameters(key,
+                out p_bn, out p_cb,
+                out q_bn, out q_cb,
+                out g_bn, out g_cb,
+                out y_bn, out y_cb,
+                out x_bn, out x_cb))
             {
-                key.DangerousAddRef(ref refAdded); // Protect access to the *_bn variables
-
-                if (!GetDsaParameters(key,
-                    out p_bn, out p_cb,
-                    out q_bn, out q_cb,
-                    out g_bn, out g_cb,
-                    out y_bn, out y_cb,
-                    out x_bn, out x_cb))
-                {
-                    throw new CryptographicException();
-                }
+                throw new CryptographicException();
```

Comment:
Ideally, all of the SafeHandles would get disposed in the if (before this throw).